test/e2e/egress_services.go

```diff
@@ -136,6 +136,65 @@ var _ = ginkgo.Describe("Egress Services", func() {
 		ginkgotable.Entry("ipv6 pods", v1.IPv6Protocol, &externalIPv6),
 	)
 
+	ginkgotable.DescribeTable("Should validate the egress SVC SNAT functionality against host-networked pods",
+		func(protocol v1.IPFamily) {
+			ginkgo.By("Creating the backend pods")
+			podsCreateSync := errgroup.Group{}
+			nodeToPod := make(map[string]string, 3)
+			for i, name := range pods {
+				name := name
+				i := i
+				podsCreateSync.Go(func() error {
+					p, err := createGenericPodWithLabel(f, name, nodes[i].Name, f.Namespace.Name, command, podsLabels)
+					framework.Logf("%s podIPs are: %v", p.Name, p.Status.PodIPs)
+					return err
+				})
+				nodeToPod[nodes[i].Name] = name
+			}
+
+			err := podsCreateSync.Wait()
+			framework.ExpectNoError(err, "failed to create backend pods")
+
+			ginkgo.By("Creating an egress service without node selectors")
+			_ = createLBServiceWithIngressIP(f.ClientSet, f.Namespace.Name, serviceName, protocol,
+				map[string]string{"k8s.ovn.org/egress-service": "{}"}, podsLabels, podHTTPPort)
+
+			ginkgo.By("Getting the IPs of the node in charge of the service")
+			egressHost, egressHostV4IP, egressHostV6IP := getEgressSVCHost(f.ClientSet, f.Namespace.Name, serviceName)
+			framework.Logf("Egress node is %s", egressHost.Name)
+
+			var dstNode v1.Node
+			hostNetPod := "host-net-pod"
+			for i := range nodes {
+				if nodes[i].Name != egressHost.Name {
+					dstNode = nodes[i]
+					break
+				}
+			}
+			ginkgo.By("Creating host-networked pod, on non-egress node to act as \"another node\"")
+			_, err = createPod(f, hostNetPod, dstNode.Name, f.Namespace.Name, []string{"/agnhost", "netexec", fmt.Sprintf("--http-port=%d", podHTTPPort)}, map[string]string{}, func(p *v1.Pod) {
+				p.Spec.HostNetwork = true
+			})
+			framework.ExpectNoError(err)
+			framework.Logf("Created pod %s on node %s", hostNetPod, dstNode.Name)
+
+			v4, v6 := getNodeAddresses(&dstNode)
+			dstIP := v4
+			expected := egressHostV4IP
+			if protocol == v1.IPv6Protocol {
+				dstIP = v6
+				expected = egressHostV6IP
+			}
+			ginkgo.By("Verifying traffic from the backend pod on egressHost should exit with their node's IP when going towards other nodes in cluster")
+			backendSrcPod := nodeToPod[egressHost.Name]
+			gomega.Consistently(func() error {
+				return curlAgnHostClientIPFromPod(f.Namespace.Name, backendSrcPod, expected, dstIP, podHTTPPort)
```

Comment:
lets loops on all pods and curl the node with the service, and ensure they all have their respective node IPs as source ip.

Reply:
ack, done.

Reply:
```
Jan 30 20:40:00.576: INFO: Skipping check for pod pod1 because its on the destination node; srcIP will be podIP
Jan 30 20:40:00.576: INFO: Running '/usr/local/bin/kubectl --server=https://127.0.0.1:44965 --kubeconfig=/home/surya/ovn.conf --namespace=egress-services-7887 exec pod2 -- /bin/sh -x -c curl -s --retry-connrefused --retry 5 --max-time 1 http://172.19.0.2:8080/clientip'
Jan 30 20:40:00.709: INFO: stderr: "+ curl -s --retry-connrefused --retry 5 --max-time 1 http://172.19.0.2:8080/clientip\n"
Jan 30 20:40:00.709: INFO: stdout: "172.19.0.4:50202"
Jan 30 20:40:00.910: INFO: Running '/usr/local/bin/kubectl --server=https://127.0.0.1:44965 --kubeconfig=/home/surya/ovn.conf --namespace=egress-services-7887 exec pod2 -- /bin/sh -x -c curl -s --retry-connrefused --retry 5 --max-time 1 http://172.19.0.2:8080/clientip'
Jan 30 20:40:01.014: INFO: stderr: "+ curl -s --retry-connrefused --retry 5 --max-time 1 http://172.19.0.2:8080/clientip\n"
Jan 30 20:40:01.014: INFO: stdout: "172.19.0.4:50206"
Jan 30 20:40:01.215: INFO: Running '/usr/local/bin/kubectl --server=https://127.0.0.1:44965 --kubeconfig=/home/surya/ovn.conf --namespace=egress-services-7887 exec pod2 -- /bin/sh -x -c curl -s --retry-connrefused --retry 5 --max-time 1 http://172.19.0.2:8080/clientip'
Jan 30 20:40:01.353: INFO: stderr: "+ curl -s --retry-connrefused --retry 5 --max-time 1 http://172.19.0.2:8080/clientip\n"
Jan 30 20:40:01.353: INFO: stdout: "172.19.0.4:50214"
Jan 30 20:40:01.553: INFO: Running '/usr/local/bin/kubectl --server=https://127.0.0.1:44965 --kubeconfig=/home/surya/ovn.conf --namespace=egress-services-7887 exec pod2 -- /bin/sh -x -c curl -s --retry-connrefused --retry 5 --max-time 1 http://172.19.0.2:8080/clientip'
Jan 30 20:40:01.673: INFO: stderr: "+ curl -s --retry-connrefused --retry 5 --max-time 1 http://172.19.0.2:8080/clientip\n"
Jan 30 20:40:01.673: INFO: stdout: "172.19.0.4:50218"
Jan 30 20:40:01.673: INFO: Running '/usr/local/bin/kubectl --server=https://127.0.0.1:44965 --kubeconfig=/home/surya/ovn.conf --namespace=egress-services-7887 exec pod3 -- /bin/sh -x -c curl -s --retry-connrefused --retry 5 --max-time 1 http://172.19.0.2:8080/clientip'
Jan 30 20:40:01.804: INFO: stderr: "+ curl -s --retry-connrefused --retry 5 --max-time 1 http://172.19.0.2:8080/clientip\n"
Jan 30 20:40:01.804: INFO: stdout: "172.19.0.3:41854"
Jan 30 20:40:02.004: INFO: Running '/usr/local/bin/kubectl --server=https://127.0.0.1:44965 --kubeconfig=/home/surya/ovn.conf --namespace=egress-services-7887 exec pod3 -- /bin/sh -x -c curl -s --retry-connrefused --retry 5 --max-time 1 http://172.19.0.2:8080/clientip'
Jan 30 20:40:02.084: INFO: stderr: "+ curl -s --retry-connrefused --retry 5 --max-time 1 http://172.19.0.2:8080/clientip\n"
Jan 30 20:40:02.084: INFO: stdout: "172.19.0.3:41870"
Jan 30 20:40:02.285: INFO: Running '/usr/local/bin/kubectl --server=https://127.0.0.1:44965 --kubeconfig=/home/surya/ovn.conf --namespace=egress-services-7887 exec pod3 -- /bin/sh -x -c curl -s --retry-connrefused --retry 5 --max-time 1 http://172.19.0.2:8080/clientip'
Jan 30 20:40:02.534: INFO: stderr: "+ curl -s --retry-connrefused --retry 5 --max-time 1 http://172.19.0.2:8080/clientip\n"
Jan 30 20:40:02.534: INFO: stdout: "172.19.0.3:41884"
```

Reply:
note that if the pod is on the same node as destination node, then we won't have srcIP as nodeIP but instead as podIP but that's out of this PR's scope so added a comment and continue in the e2e test.